envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/NodeMetadataValidator.kt

```diff
@@ -55,4 +61,17 @@ class NodeMetadataValidator(
 
     private fun isAllowedToHaveAllServiceDependencies(metadata: NodeMetadata) = properties
         .outgoingPermissions.servicesAllowedToUseWildcard.contains(metadata.serviceName)
+
+    private fun validateConfigurationMode(metadata: NodeMetadata) {
+        if (!properties.configurationMode.ads && !properties.configurationMode.xds) {
+            throw ConfigurationModeNotSupportedException(metadata.serviceName, "Neither ADS nor XDS")
+        }
+        if (metadata.ads && !properties.configurationMode.ads) {
+            throw ConfigurationModeNotSupportedException(metadata.serviceName, "ADS")
+        }
+        if (!metadata.ads && !properties.configurationMode.xds) {
+            throw ConfigurationModeNotSupportedException(metadata.serviceName, "XDS")
```

Comment:
Maybe also log this incident on WARN level? I am not sure if validation exceptions are logged in EC. If they are, then nevermind.

Reply:
nvm, I just checked it is logged in EC